### PR TITLE
added auto-register config option

### PR DESCRIPTION
**Pull request by rafaelhbarros**
related to #231

### CODE REVIEW

File: binstar_client/commands/config.py [hunk 1]
```diff
@@ -50,6 +50,13 @@
   * `url`: Set the anaconda api url (default: https://api.anaconda.org)
   * `verify_ssl`: Perform ssl validation on the https requests.
     verify_ssl may be `True`, `False` or a path to a root CA pem file.
+
+
+###### Toggle auto-register when doing anaconda upload
+If turned off, whenever an upload is done with a new package name, the upload is going to fail
```
Comment (srossross) [inline binstar_client/commands/config.py]: how about re-wording?
```
The default is yes. If `no`, then an upload will fail if the package name does not already exist on the server.
```

File: binstar_client/commands/config.py [hunk 2]
```diff
@@ -50,6 +50,13 @@
...
+
+  anaconda config --set auto_register yes|no
```
Comment (dsludwig) [inline binstar_client/commands/config.py]: I believe 4 spaces indent is required for code blocks to be recognized by the docs site.

Comment (rafaelhbarros) [inline binstar_client/commands/config.py]: but like, the block above this one starts with an empty line and follows with a first column char.

Comment (dsludwig) [inline binstar_client/commands/config.py]: https://daringfireball.net/projects/markdown/syntax#precode

> To produce a code block in Markdown, simply indent every line of the
block by at least 4 spaces or 1 tab.

Comment (rafaelhbarros) [inline binstar_client/commands/config.py]: oh the code block.

**Comment by rafaelhbarros** [comment]
Tests on linux & windows pass.

**Comment by dsludwig** [comment]
@rafaelhbarros Do you want to update to use this text? It is a bit clearer

> The default is `yes`, automatically create a new package when uploading. If `no`, then an upload will fail if the package name does not already exist on the server.

**Comment by rafaelhbarros** [comment]
@dsludwig done, good idea.

**Comment by dsludwig** [comment]
LGTM

File: binstar_client/commands/upload.py [hunk 3]
```diff
@@ -258,8 +260,13 @@ def add_parser(subparsers):
...
     mgroup.add_argument('--thumbnail', help='Notebook\'s thumbnail image')
 
-    parser.add_argument("--no-register", action="store_true", default=False,
+    register_group = parser.add_mutually_exclusive_group()
+    register_group.add_argument("--no-register", dest="auto_register", action="store_false",
                         help='Don\'t create a new package namespace if it does not exist')
+    register_group.add_argument("--register", dest="auto_register", action="store_true",
+                        help='Create a new package namespace if it does not exist',
+                        default=bool(get_config().get('auto_register')))
+    parser.set_defaults(auto_register=bool(get_config().get('auto_register', True)))
```
Comment (dsludwig) [inline binstar_client/commands/upload.py]: does this default for auto_register need to be set twice?

Comment (rafaelhbarros) [inline binstar_client/commands/upload.py]: no.

Comment (rafaelhbarros) [inline binstar_client/commands/upload.py]: changed.

**Pull request merged**